### DISCUSSION TITLE
can kong support custom nginx location dynamic configure

### DISCUSSION THREAD

**Discussion by ly123-liu** [Kong/kong#7593] [Help]
kong version: 2.4.1

can kong provide server snippet func like nginx-ingress ?  since sometimes we has custom location to add , such as static location, if else statement  ..etc  , is there plan to do that ? thanks 

or how can i do that in kong ? please help me ,thanks

**Comment by Tieske** [comment]
you can create a custom nginx config template see https://docs.konghq.com/gateway-oss/2.5.x/configuration/#serving-both-a-website-and-your-apis-from-kong

**Comment by ly123-liu** [comment]
nginx.template can only be used on the startup, after that ,how can we dynamic add locations ?

**Reply by zffocussss** [reply]
> nginx.template can only be used on the startup, after that ,how can we dynamic add locations ?

hi liu:
   in this case,you can try this way
1.use [autoreload](https://github.com/stakater/Reloader) in your K8s cluster
2.refer to custom template as configmap,and mount to pods

**Comment by Tieske** [comment]
You can't. In that case you'll have to use regular Kong routes and services

**Comment by locao** [comment]
Hi @ly123-liu! Can you explain what you are trying to achieve?
